src/OmniSharp.Roslyn/OmniSharpWorkspace.cs

```diff
@@ -427,5 +474,23 @@ public void RemoveAdditionalDocument(DocumentId documentId)
         {
             OnAdditionalDocumentRemoved(documentId);
         }
+
+        protected override void ApplyProjectChanges(ProjectChanges projectChanges)
+        {
+            // since Roslyn currently doesn't handle DefaultNamespace changes via ApplyProjectChanges
+            // and OnDefaultNamespaceChanged is internal, we use reflection for now
+            if (projectChanges.NewProject.DefaultNamespace != projectChanges.OldProject.DefaultNamespace)
+            {
+                var onDefaultNamespaceChanged = this.GetType().GetMethod("OnDefaultNamespaceChanged", BindingFlags.Instance | BindingFlags.NonPublic);
+                if (onDefaultNamespaceChanged != null)
+                {
+                    onDefaultNamespaceChanged.Invoke(this, new object[] { projectChanges.ProjectId, projectChanges.NewProject.DefaultNamespace });
+                }
+            }
+            else
+            {
+                base.ApplyProjectChanges(projectChanges);
```

Comment:
Should we call the base `ApplyProjectChanges` all the time or only if the namespace didn't change?  Would this possibly miss some changes if they were batched together with the namespace change?

Reply:
thanks @david-driscoll this is a great catch - change accordingly 👍